gobblin-utility/src/main/java/gobblin/util/executors/IteratorExecutor.java

```diff
@@ -87,4 +105,56 @@ public IteratorExecutor(Iterator<Callable<T>> runnableIterator, int numThreads,
     return futures;
   }
 
+  /**
+   * Execute the tasks in the task {@link Iterator}. Blocks until all tasks are completed. Gets the results of each
+   * task, and for each task returns either its result or the thrown {@link ExecutionException}.
+   *
+   * @return a list containing for each task either its result or the {@link ExecutionException} thrown, in the same
+   *        order as the input {@link Iterator}.
+   * @throws InterruptedException
+   */
+  public List<Either<T, ExecutionException>> executeAndGetResults() throws InterruptedException {
+    List<Either<T, ExecutionException>> results = Lists.newArrayList();
+    List<Future<T>> futures = execute();
+    for (Future<T> future : futures) {
+      try {
+        results.add(Either.<T, ExecutionException>left(future.get()));
+      } catch (ExecutionException ee) {
+        results.add(Either.<T, ExecutionException>right(ee));
+      }
+    }
+    return results;
+  }
+
+  /**
+   * Utility method that checks whether all tasks succeeded from the output of {@link #executeAndGetResults()}.
+   * @return true if all tasks succeeded.
+   */
+  public static <T> boolean verifyAllSuccessful(List<Either<T, ExecutionException>> results) {
+    return Iterables.all(results, new Predicate<Either<T, ExecutionException>>() {
+      @Override
+      public boolean apply(@Nullable Either<T, ExecutionException> input) {
+        return input instanceof Either.Left;
+      }
+    });
+  }
+
+  /**
+   * Utility method unpacks the {@link Either} results from {@link IteratorExecutor} to the raw results of the generic
+   * type.
+   * @return List of {@link Callable} results.
+   * @throws ExecutionException if any {@link Callable} failed.
+   */
+  public static <T> List<T> unpackResults(List<Either<T, ExecutionException>> results) throws ExecutionException {
```

Comment:
Is this used? Also, seems to me that the real content of `results` can be left for users to handle.